relay-server/src/actors/project_cache.rs

```diff
@@ -467,8 +506,18 @@ impl ProjectCacheBroker {
             no_cache,
         } = message;
 
+        let invalid = state.invalid();
+
         self.get_or_create_project(project_key)
-            .update_state(state, no_cache)
+            .update_state(state, no_cache);
+
+        // There is no much sense to dequeue and check anything if the incoming state is invalid.
+        if !invalid {
+            let mut envelopes = self.spool.dequeue(project_key);
+            while let Some((envelope, envelope_context)) = envelopes.pop_front() {
+                self.handle_processing(envelope, envelope_context)
```

Comment:
This puts an envelope back on the queue when one of the two project states is still missing. I think this works, but it would be neat to have an optimization to make sure that an envelope gets spooled only once, and retrieved only when both project states are present.

Reply:
This would also prevent infinite looping, although that might be only a theoretical concern.